lib/util/pssh.js

```diff
@@ -189,4 +216,36 @@ shaka.util.Pssh = class {
 
     return shaka.util.Uint8ArrayUtils.concat(...dedupedInitDatas);
   }
+
+  /**
+   * Get a pssh from the given keySystem and moov.
+   *
+   * @param {!Uint8Array} initData
+   * @param {string} keySystem
+   * @return {!Uint8Array}
+   */
+  static getPsshFromInitData(initData, keySystem) {
+    const systemId = shaka.util.Pssh.defaultSystemIds_.get(keySystem);
+    const psshInfo = new shaka.util.Pssh(initData, true);
+
+    const index = psshInfo.systemIds.indexOf(systemId);
+    // If the index is -1 we can not found the correct pssh
+    if (index === -1) {
+      return new Uint8Array(0);
+    }
+
+    const pssh = psshInfo.data[index];
+    if (!pssh) {
+      return new Uint8Array(0);
+    }
```

Comment:
You don't handle these error case at the call site.  So if the first one we look for isn't found, it will send blank init data to DrmEngine and try to generate a license request, which will fail.

Reply:
Do you think it is correct to throw the error `ENCRYPTED_CONTENT_WITHOUT_DRM_INFO`?

Reply:
In this case, we launch shaka.util.Error.Code.ENCRYPTED_CONTENT_WITHOUT_DRM_INFO